htdocs/kernel/object.php

```
@@ -1066,6 +1066,43 @@ public function __construct(XoopsDatabase $db)
+     * @param  string   $defaultColumn  column used when no valid clause remains
+     * @return string   a safe ORDER BY column list including directions
+     */
+    public static function buildOrderBy($sort, $order, array $allowedColumns, $defaultColumn)
+    {
+        $order = strtoupper(trim((string) $order));
+        if ('ASC' !== $order && 'DESC' !== $order) {
+            $order = 'ASC';
+        }
+        $parts = [];
+        foreach (explode(',', (string) $sort) as $token) {
+            if (preg_match('/^\s*((?:[A-Za-z_]\w*\.)?)([A-Za-z_]\w*)\s*(ASC|DESC)?\s*$/i', $token, $m)
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $m. Configured minimum length is 3. (undefined)

(ShortVariable)